test-basic/documents-quick.js

```diff
@@ -167,3 +167,67 @@ describe('quick path', function(){
     .catch(done);
   });
 });
+
+describe('test connection', function(){
+ var assert = require('assert');
+    it('should test the connection', function(done){
+     	var flag = false;
+     db.testConnection()
+      .result(function(response) {
+        flag = true;
+     }).then(function(){
+        assert(flag == true);
```

Comment:
Because the result() callback doesn't return a promise, I believe the then() callback will never execute.

Same comment on subsequent tests.